Sources/JWTKit/ECDSA/ECDSAKey.swift

```diff
@@ -190,3 +199,197 @@ extension ECDSAKey.Curve {
         }
     }
 }
+
+public protocol ECDSACurveType: Sendable {
+    static var curve: ECDSAKey.Curve { get }
+}
+
+extension P256: ECDSACurveType, @unchecked @retroactive Sendable {
+    static public var curve: ECDSAKey.Curve { .p256 }
+}
+public typealias ES256PublicKey = ECDSA.PublicKey<P256>
+public typealias ES256PrivateKey = ECDSA.PrivateKey<P256>
+
+extension P384: ECDSACurveType, @unchecked @retroactive Sendable {
+    static public var curve: ECDSAKey.Curve { .p384 }
+}
+public typealias ES384PublicKey = ECDSA.PublicKey<P384>
+public typealias ES384PrivateKey = ECDSA.PrivateKey<P384>
+
+extension P521: ECDSACurveType, @unchecked @retroactive Sendable {
+    static public var curve: ECDSAKey.Curve { .p521 }
+}
+public typealias ES512PublicKey = ECDSA.PublicKey<P521>
+public typealias ES512PrivateKey = ECDSA.PrivateKey<P521>
+
+public enum ECDSA: Sendable {
+    /// ECDSA.PublicKey was introduced in v5 and replaces ``ECDSAKey``.
+    ///
+    /// - Note: Please migrate over to ``ECDSA/PublicKey`` before updating to v5, though if you plan on remaining on v4, ``ECDSAKey`` can continue to be used.
+    public struct PublicKey<Curve: ECDSACurveType> {
+        let key: ECDSAKey
+        init(key: ECDSAKey) { self.key = key }
```

Comment:
These are wrappers on the existing keys to support the new types a user would be forced into, wile not adding much new functionality. I added bare minimum support for all ECDSA public and private keys.